pkg/rkt-stage1/0006-Enable-local-networking-in-containers.patch

```diff
@@ -0,0 +1,164 @@
+ 
+ # Clean the repo, but save the vendor area
+diff --git a/files/run b/files/run
+index d001b5a..fbb3a71 100755
+--- a/files/run
++++ b/files/run
+@@ -12,6 +12,7 @@ gw=""
+ route=""
+ pvcalls=0
+ bridge="xenbr0"
++dhcp=1
+ 
+ for i in $*
+ do
+@@ -63,7 +64,13 @@ echo "p9=[ 'tag=share_dir,security_model=none,path=$mountpoint' ]" >> $outconfig
+ 
+ if [ -f "$STAGE1_SEED_XL_CFG" ] ; then
+     grep -Ev '^(disk|bootloader|root|extra)' < "$STAGE1_SEED_XL_CFG" >> $outconfig
+-    echo extra=\'console=hvc0 root=9p\' >> $outconfig
++    if test $dhcp -eq 0
++    then
++        echo extra=\'console=hvc0 root=9p\' >> $outconfig
++    else
++        cp $stage1/udhcpc_script.sh $stage2
++        echo extra=\'console=hvc0 root=9p dhcp=1\' >> $outconfig
++    fi
+ else	
+     echo "memory = 1024" >> $outconfig
+     echo "vcpus = 2" >> $outconfig
+diff --git a/files/udhcpc_script.sh b/files/udhcpc_script.sh
+new file mode 100755
+index 0000000..3b8c2e6
+--- /dev/null
++++ b/files/udhcpc_script.sh
+@@ -0,0 +1,78 @@
++#!/bin/busybox sh
++
++[ -z "$1" ] && echo 'Error: should be called from udhcpc' && exit 1
++
++# create etc directory if not already done
++mkdir -p /mnt/rootfs/etc
++
++# save config information for $interface
++CFG="/mnt/rootfs/etc/udhcpc.${interface}.cfg"
++
++RESOLV_CONF='/mnt/rootfs/etc/resolv.conf'
++
++# interface for which DNS is to be configured
++PEERDNS_IF=eth0
++
++case "$1" in
++  deconfig)
++    echo "udhcpc op deconfig interface ${interface}"
++    # bring interface up, but with no IP configured
++    ip addr flush dev $interface
++    ip link set $interface up
++    # remove any stored config info for this $interface
++    rm -f $CFG
++    # remove previous dns
++    rm -f $RESOLV_CONF
++    ;;
++  bound)
++    echo "udhcpc op bound interface ${interface}"
++    # save config info for $interface
++    set > $CFG
++    # configure interface and routes
++    ip addr flush dev $interface
++    ip addr add ${ip}/${mask} dev $interface
++    [ -n "$router" ] && ip route add default via ${router%% *} dev $interface
++    # setup dns
++    if [ "$interface" == "$PEERDNS_IF" ] ; then
++      [ -n "$domain" ] && echo search $domain > $RESOLV_CONF
++      for i in $dns ; do
++        echo nameserver $i >> $RESOLV_CONF
++      done
++    fi
++    ;;
++  renew)
++    echo "udhcpc op renew interface ${interface}"
++    # compare new vs. previous config info:
++    set > ${CFG}.new
++    for i in $(diff -U1 $CFG ${CFG}.new | grep -E ^[+-] \
++                                        | tail +3 \
++                                        | awk -F[+-=] '{print $2}') ; do
++      case "$i" in
++        ip|mask|router)
++          REDO_NET='yes'
++          ;;
++        domain|dns)
++          REDO_DNS='yes'
++          ;;
++      esac
++    done
++    # save new config info:
++    mv -f ${CFG}.new $CFG
++    # make only necessary changes, as per config comparison:
++    if [ -n "$REDO_NET" ] ; then
++      ip addr flush dev $interface
```

Comment:
According to the udhcp documentation the IP address will never change when you get a renew. So these two lines are not needed.

Reply:
Ah, but the subnet mask can change. I guess that means you need to delete and re-add the ip addr.